src/crimson/net/ProtocolV1.cc

```diff
@@ -777,27 +777,28 @@ seastar::future<> ProtocolV1::read_message()
       ::decode(m.footer, p);
       auto msg = ::decode_message(nullptr, 0, m.header, m.footer,
                                   m.front, m.middle, m.data, nullptr);
-      if (!msg) {
-	logger().debug("decode message failed");
-	return;
+      if (unlikely(!msg)) {
+        logger().warn("{} decode message failed", conn);
+        throw std::system_error{make_error_code(error::negotiation_failure)};
```

Comment:
can we define a new error for corrupted message?

Reply:
done.